src/ploigos_step_runner/step_implementers/automated_governance/rekor.py

```diff
@@ -0,0 +1,199 @@
+"""`StepImplementer` for the `automated-governance` step using Rekor.
+
+Step Configuration
+------------------
+Step configuration expected as input to this step.
+Could come from:
+
+  * static configuration
+  * runtime configuration
+  * previous step results
+
+Configuration Key | Required? | Default        | Description
+------------------|-----------|----------------|-----------
+`rekor-server`     | True |         | Version to use when building the container image
+
+Result Artifacts
+----------------
+Results artifacts output by this step.
+
+Result Artifact Key | Description
+--------------------------|------------
+`container-image-version` | Container version to tag built image with
+`image-tar-file`          | Path to the built container image as a tar file
+`image-tar-hash`          | Path to the built container image as a tar file
+"""
+
+import os
+import pprint
+import sh
+import re
+import json
+import base64
+import textwrap
+import subprocess
+import hashlib
+
+from pathlib import Path
+import sys
+from io import StringIO
+from contextlib import redirect_stderr, redirect_stdout
+from ploigos_step_runner import StepImplementer, StepResult, WorkflowResult
+from ploigos_step_runner.utils.io import create_sh_redirect_to_multiple_streams_fn_callback
+from ploigos_step_runner.utils.io import TextIOIndenter
+from ploigos_step_runner.utils.dict import deep_merge
+from ploigos_step_runner.utils.file import base64_encode
+from ploigos_step_runner.utils.file import get_file_hash
+from ploigos_step_runner.utils.gpg import get_gpg_key
+
+
+DEFAULT_CONFIG = {
+    'rekor-server': '',
+    'gpg-key': '',
+    'gpg-user': ''
+}
+
+REQUIRED_CONFIG_OR_PREVIOUS_STEP_RESULT_ARTIFACT_KEYS = [
+    'rekor-server',
+    'gpg-key',
+    'gpg-user'
+]
+
+
+class Rekor(StepImplementer):  # pylint: disable=too-few-public-methods
+    """`StepImplementer` for the `automated-governance` step using Rekor.
+    """
+
+    @staticmethod
+    def step_implementer_config_defaults():
```

Comment:
you can just get rid of this funciton since we dont have any default config and the parent class will return an empty array.